src/main/java/core/basesyntax/MatrixOperation.java

```diff
@@ -1,19 +1,65 @@
 package core.basesyntax;
 
 /**
- * Реалізуйте методи для знаходження головної діагоналі прямокутної матриці,
- * максимального та мінімального елементів.
+ * Реалізуйте методи для знаходження головної діагоналі прямокутної матриці, максимального та
+ * мінімального елементів.
  */
 public class MatrixOperation {
+
+    /**
+     * Returns main diagonal of matrix.
+     *
+     * @return main diagonal of matrix
+     */
     public int[] getDiagonal(int[][] matrix) {
-        return null;
+        int[] mainDiagonal = new int[matrix.length];
+        for (int i = 0; i < matrix.length; i++) {
+            mainDiagonal[i] = matrix[i][i];
+        }
+        return mainDiagonal;
     }
 
+    /**
+     * Returns min value of matrix.
+     *
+     * @return min value of matrix, or 0 if matrix empty
+     */
     public int getMin(int[][] matrix) {
-        return 0;
+        int min;
```

Comment:
we can simplify:
        if (matrix == null) {
            return 0;
        } 
int min = matrix[0][0];else {
    ...